policy-controller/k8s/index/src/pod.rs

```diff
@@ -85,13 +85,15 @@ impl Index {
             .namespaces
             .get_or_default(pod.namespace().expect("namespace must be set"));
 
-        pods.apply(
+        if let Err(error) = pods.apply(
             pod,
             servers,
             &mut self.lookups,
             *default_policy,
             &mut self.default_policy_watches,
-        )
+        ) {
+            warn!(?error, "failed to apply pod");
```

Comment:
nit: `PodIndex::apply` returns an `anyhow::Result` --- so the error type should implement `Display` as well:
```suggestion
            warn!(%error, "failed to apply pod");
```